src/compiler/scala/tools/nsc/typechecker/RefChecks.scala

```diff
@@ -1405,18 +1405,45 @@ abstract class RefChecks extends Transform {
       }
     }
 
-    private def checkAnnotations(tpes: List[Type], tree: Tree) = tpes foreach { tp =>
-      checkTypeRef(tp, tree, skipBounds = false)
-      checkTypeRefBounds(tp, tree)
-    }
-    private def doTypeTraversal(tree: Tree)(f: Type => Unit) = if (!inPattern) tree.tpe foreach f
+    private def applyRefchecksToAnnotations(tree: Tree): Tree = {
```

Comment:
Why does this method return a tree now? It seems all callsites are in statement position.

Reply:
I had thought I may have needed to return a different tree, rather than just mutating it. I was wrong. It's `Unit`ary now.